src/player.cpp

```diff
@@ -7929,75 +7937,52 @@ class ma_style_callback : public uimenu_callback
 
 bool player::pick_style() // Style selection menu
 {
-    //Create menu
-    // Entries:
-    // 0: Cancel
-    // 1: No style
-    // x: dynamic list of selectable styles
+    enum style_selection {
+        KEEP_HANDS_FREE = 0,
+        STYLE_OFFSET
+    };
 
-    //If there are style already, cursor starts there
+    // If there are style already, cursor starts there
     // if no selected styles, cursor starts from no-style
 
     // Any other keys quit the menu
+    std::vector<matype_id> selectable_styles = {{
+        style_none, style_kicks
+    }};
+    const std::vector<matype_id> &real_styles = has_active_bionic( bio_cqb ) ? bio_cqb_styles : ma_styles;
+    selectable_styles.insert( selectable_styles.end(), real_styles.begin(), real_styles.end() );
 
     uimenu kmenu;
+    kmenu.return_invalid = true;
     kmenu.text = _("Select a style (press ? for more info)");
-    std::unique_ptr<ma_style_callback> ma_style_info(new ma_style_callback());
+    std::unique_ptr<ma_style_callback> ma_style_info( new ma_style_callback( ( size_t )STYLE_OFFSET, selectable_styles ) );
     kmenu.callback = ma_style_info.get();
     kmenu.desc_enabled = true;
-    kmenu.addentry( 0, true, 'c', _("Cancel") );
-    if (keep_hands_free) {
-      kmenu.addentry_desc( 1, true, 'h', _("Keep hands free (on)"), _("When this is enabled, player won't wield things unless explicitly told to."));
-    }
-    else {
-      kmenu.addentry_desc( 1, true, 'h', _("Keep hands free (off)"), _("When this is enabled, player won't wield things unless explicitly told to."));
-    }
+    kmenu.addentry_desc( KEEP_HANDS_FREE, true, 'h', keep_hands_free ? _( "Keep hands free (on)" ) : _( "Keep hands free (off)" ),
+                         _( "When this is enabled, player won't wield things unless explicitly told to." ) );
 
-    if (has_active_bionic( bio_cqb ) ) {
-        for(size_t i = 0; i < bio_cqb_styles.size(); i++) {
-            if( bio_cqb_styles[i].is_valid() ) {
-                auto &style = bio_cqb_styles[i].obj();
-                kmenu.addentry_desc( i + 2, true, -1, style.name, style.description );
-            }
-        }
+    kmenu.selected = STYLE_OFFSET;
+    kmenu.return_invalid = true; //cancel with any other keys
```

Comment:
This is set twice (first at line 7956).

Also: if you want to remove hacks, remove the pointer indirection from `ma_style_info`, make it a normal stack allocated object of type `ma_style_callback` and set `kmenu.callback` to its address. It needs to live as long as the menu lives, which is not dynamically allocated either.